corehq/form_processor/submission_context.py

```diff
@@ -0,0 +1,43 @@
+import lxml.etree
+from memoized import memoized
+import xml2json
+
+from corehq.form_processor.extension_points import get_form_submission_context_class
+from corehq.form_processor.utils.metadata import scrub_meta
+
+from couchforms.util import legacy_notification_assert
+
+
+class SubmissionFormContext(object):
+    def __init__(self, submission_post):
+        self.submission_post = submission_post
+        self._instance = submission_post.instance
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        pass
+
+    def get_instance_xml(self):
+        try:
+            return xml2json.get_xml_from_string(self._instance)
+        except xml2json.XMLSyntaxError:
+            return None
+
+    def get_instance(self):
+        return self._instance
+
+    def update_instance(self, xml):
+        self._instance = lxml.etree.tostring(xml)
+
+    def post_process_form(self, xform):
+        self.submission_post.set_submission_properties(xform)
```

Comment:
why not just move all this code into the context? (also `invalidate_caches`)

Reply:
There is a lot in those methods that take values on `SubmissionPost` object like submit_ip, path etc which would need to be moved as well here, putting weight on this class. 

I think having this class to just manage steps would be nice and clean and If we really need to trim down SubmissionPost, it would be better to get its responsibilities out but to a different/new class.
Let me know if you think otherwise?

Reply:
I don't know, this is adding a fair amount more complexity to an already complex process:
 view -> function -> class -> context class. I think the less back and forth we have the better. One option would be to create the context outside of `SubmissionPost` and pass it in. That would move all those fields out of `SubmissionPost` and into the context class.

Reply:
Cool.
Passing context would need change everywhere SubmissionPost is initialised to just instantiated it in SubmissionPost itself, 
https://github.com/dimagi/commcare-hq/pull/28409/commits/1acf5964518aeca250158983f5ce8f2a46244554

I think SubmissionContext made sense as the name when we using it as context, I was thinking of FormSubmissionProcessor but we do have FormProcessor and similar names across the code already. Any suggestions on improving that?